packages/core/src/main.ts

```diff
@@ -0,0 +1,144 @@
+import {
+  ConnectedStarknetWindowObject,
+  StarknetWindowObject,
+} from "./StarknetWindowObject"
+import discovery, { WalletProvider } from "./discovery"
+import { IStorageWrapper, LocalStorageWrapper } from "./localStorageStore"
+import { pipe } from "./utils"
+import { FilterList, filterBy, filterByPreAuthorized } from "./wallet/filter"
+import { isWalletObj } from "./wallet/isWalletObject"
+import { scanObjectForWallets } from "./wallet/scan"
+import { Sort, sortBy } from "./wallet/sort"
+
+export type {
+  AccountChangeEventHandler,
+  AddStarknetChainParameters,
+  ConnectedStarknetWindowObject,
+  NetworkChangeEventHandler,
+  RpcMessage,
+  StarknetWindowObject,
+  SwitchStarknetChainParameter,
+  WalletEventHandlers,
+  WalletEvents,
+  WatchAssetParameters,
+} from "./StarknetWindowObject"
+export type { WalletProvider } from "./discovery"
+
+export interface SnWalletOptions {
+  windowObject: Record<string, any>
+  isWalletObject: (wallet: any) => boolean
+  storageFactoryImplementation: (name: string) => IStorageWrapper
+}
+
+const defaultOptions: SnWalletOptions = {
+  windowObject: window,
+  isWalletObject: isWalletObj,
+  storageFactoryImplementation: (name: string) => new LocalStorageWrapper(name),
+}
+
+export interface GetWalletOptions {
+  sort?: Sort
+  include?: FilterList
+  exclude?: FilterList
+}
+
+interface SnWalletResult {
+  getAvailableWallets: (
+    options?: GetWalletOptions,
+  ) => Promise<StarknetWindowObject[]> // Returns all wallets available in the window object
+  getPreAuthorizedWallets: (
+    options?: GetWalletOptions,
+  ) => Promise<StarknetWindowObject[]> // Returns only preauthorized wallets available in the window object
+  getDiscoveryWallets: (options?: GetWalletOptions) => Promise<WalletProvider[]> // Returns all wallets in existence (from discovery file)
+  getLastConnectedWallet: () => Promise<StarknetWindowObject | null> // Returns the last wallet connected when it's still connected
+  getDefaultWallet: () => Promise<StarknetWindowObject | null> // Returns the default wallet
```

Comment:
we'll need `| undefined` for backward compatibility (we can have both `| null | undefined`)